### ISSUE TITLE
Reverse normalize on batches of images?(tensor type not PIL image)

### ISSUE THREAD

**Issue by Charlie4zc** [pytorch/vision#1074]
I have batches of images with size of `(B,C,H,W)` (torch.tensor) i want to unnormalized it with mean/std tensor (e.g.:[0.4,0.3,0.2]) with size of`(C,H,W)`

My question is is there any operations in Pytorch can do this like` dataset.map` or  `tf.expand_dims` in Tensorflow ?

Be careful ,it not PIL image format

**Comment by pmeier** [comment]
You can achieve that by adding a batch dimension and let the [broadcasting mechanics](https://pytorch.org/docs/stable/notes/broadcasting.html) handle it for you:

```python
mean, std = mean.unsqueeze(0), std.unsqueeze(0) # 1xCxHxW
y = x * std + mean
```

---

Why do your normalisation tensors (`mean` and `std`) have spatial dimensions (`H` and `W`)? This would indicate that the `mean` and `std` change depending on the position in the image, which is quite unusual. I'm guessing you want to do a per-channel normalisation, which makes `mean` and `std` 1D tensors of size `C`.

To follow up on you example:

```python
device, dtype = x.device, x.dtype
mean = torch.tensor([0.4, 0.3, 0.2], device=device, dtype=dtype).view(1, -1, 1, 1)
std = torch.tensor([0.4, 0.3, 0.2], device=device, dtype=dtype).view(1, -1, 1, 1)
y = x * std + mean
```

If you want to have the denormalisation as a `torchvision` transform have a look at [this comment](https://github.com/pytorch/vision/issues/848#issuecomment-482547380) from a similar issue.

**Comment by Charlie4zc** [comment]
Thank you so much!
